data-prepper-api/src/main/java/org/opensearch/dataprepper/typeconverter/DoubleConverter.java

```diff
@@ -13,12 +14,15 @@ public Double convert(Object source) throws IllegalArgumentException {
         if (source instanceof Double) {
             return (Double)source;
         }
-        if (source instanceof Number) {
+        if ((source instanceof Number) && !(source instanceof BigDecimal)) {
             return (double)(((Number)source).intValue());
         }
         if (source instanceof Boolean) {
             return (double)(((Boolean)source) ? 1.0 : 0.0);
         }
+        if (source instanceof BigDecimal) {
```

Comment:
I wonder if we can combine all Number type under:
```java
        if ((source instanceof Number)) {
            return (((Number)source).doubleValue());
        }
```

Not sure why Number is converted to intValue first.

Reply:
Acknowledged, updated it.